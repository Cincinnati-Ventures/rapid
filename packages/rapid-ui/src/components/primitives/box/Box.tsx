import React from 'react';
import { RapidStyles } from '../../../utils';

type BoxType = 'div';

interface BoxProps extends React.HTMLAttributes<HTMLDivElement> {
	styles?: string;
	as?: BoxType;
}

const THEME_CLASSNAME = 'rapid-box';

const Box = React.forwardRef<HTMLDivElement, BoxProps>(
	({ styles, as = 'div', ...rest }, ref) => {
		const Component = (type: BoxType) => {
			switch (type) {
				case 'div':
					return (
						<div
							ref={ref}
							className={RapidStyles(
								styles || rest.className,
								'', //defaultStyles
								THEME_CLASSNAME,
							)}
						/>
					);
			}
		};
		return Component(as);
	},
);

Box.displayName = 'Box';

export default Box;
