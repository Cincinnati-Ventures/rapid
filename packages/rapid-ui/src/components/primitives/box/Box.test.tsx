import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Box from "./Box";

describe("Box", () => {
  it("should render default correctly", () => {
    const { container } = render(<Box />);
    expect(container).toMatchSnapshot();
  });
});
