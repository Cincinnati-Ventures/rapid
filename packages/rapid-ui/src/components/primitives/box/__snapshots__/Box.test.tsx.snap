// Vitest Snapshot v1

exports[`Box > should render default correctly 1`] = `
<div>
  <div
    class="rapid-box"
  />
</div>
`;
